js/ui/status/missionGame.js

```diff
@@ -0,0 +1,528 @@
+    return dst;
+}
+
+const ScrolledLabel = new Lang.Class({
+    Name: 'ScrolledLabel',
+    Extends: St.Label,
+
+    _init: function(params) {
+        let parentParams = copyObject(params);
+        parentParams.text = '';
+
+        this.parent(parentParams);
+        this._text = params.text;
+        this._textIndex = 0;
+        this._scrollTimer = 0;
+        this._waitCount = 0;
+        this.complete = false;
+
+        this.style_class = 'chatbox-character-text';
+
+        this.set_x_expand(true);
+        this.clutter_text.set_x_expand(true);
+    },
+    start: function(scrollView) {
+        /* Avoid double-start */
+        if (this._scrollTimer) {
+            return;
+        }
+
+        /* Immediately display the first character so that
+         * scrolling to the bottom will work */
+        this._textIndex++;
+        this.set_text(this._text.slice(0, this._textIndex));
+
+        /* Add a timeout to gradually display all this text */
+        this._scrollTimer = GLib.timeout_add(GLib.PRIORITY_DEFAULT, 10, Lang.bind(this, function() {
+            if (this._waitCount > 0) {
+                this._waitCount--;
+                return true;
+            }
+
+            if (this._textIndex === this._text.length) {
+                this.complete = true;
+                this.emit('finished-scrolled');
+                return false;
+            }
+
+            this._textIndex++;
+            this.set_text(this._text.slice(0, this._textIndex));
+
+            /* Stop on punctuation */
+            if ("!?.".indexOf(this._text[this._textIndex - 1]) !== -1) {
+                this._waitCount = 100;
+                return true;
+            }
+
+            /* Running this on a timer every time is not
+             * ideal, but I haven't found a better way
+             * ensure the view is reliably always
+             * scrolled */
+            scrollView();
+
+            return true;
+        }));
+    },
+    fastForward: function() {
+        this._textIndex = this._text.length - 1;
+    }
+});
+Signals.addSignalMethods(ScrolledLabel.prototype);
+
+WrappedLabel = new Lang.Class({
+    Name: 'WrappedLabel',
+    Extends: ScrolledLabel,
+    _init: function(params) {
+        this.parent(params);
+        this._text = wrapTextWith(this._text, WRAP_CONSTANT, "> ").join("\n");
+        this.fastForward();
+    }
+});
+
+const UserResponseLabel = new Lang.Class({
+    Name: 'UserResponseLabel',
+    Extends: St.Label,
+
+    _init: function(params) {
+        let parentParams = copyObject(params);
+        parentParams.text = '';
+
+        this.parent(parentParams);
+        this._text = params.text;
+        this.complete = false;
+
+        this.style_class = 'chatbox-user-text';
+
+        this.set_x_expand(true);
+        this.clutter_text.set_x_expand(true);
+    },
+    start: function(scrollView) {
+        this.set_text(this._text);
+        this["translation-y"] = 5;
+        this["opacity"] = 0;
```

Comment:
This class was dropped in fe989841d2cba73205f1e051be9dbb5ee416bc17 so no longer necessary :)